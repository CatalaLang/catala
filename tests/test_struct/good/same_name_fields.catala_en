## Article

```catala
declaration structure Foo:
  data f content integer

declaration structure Bar:
  data f content integer

declaration scope A:
  context output x content Foo
  context output y content integer

scope A:
  definition x equals Foo { -- f: 1 }
  definition y equals x.Foo.f
```



```catala-test-inline
$ catala Typecheck --check_invariants
[WARNING] The structure "Bar" is never used; maybe it's unnecessary?

┌─⯈ tests/test_struct/good/same_name_fields.catala_en:7.23-7.26:
└─┐
7 │ declaration structure Bar:
  │                       ‾‾‾
  └─ Article
[RESULT]
Invariant Dcalc__Invariants.invariant_typing_defaults
   checked. result: [39/39]
[RESULT]
Invariant Dcalc__Invariants.invariant_match_inversion
   checked. result: [0/0]
[RESULT]
Invariant Dcalc__Invariants.invariant_app_inversion
   checked. result: [2/2]
[RESULT]
Invariant Dcalc__Invariants.invariant_no_return_a_function
   checked. result: [0/0]
[RESULT]
Invariant Dcalc__Invariants.invariant_no_partial_evaluation
   checked. result: [2/2]
[RESULT]
Invariant Dcalc__Invariants.invariant_default_no_arrow
   checked. result: [6/6]
[RESULT] Typechecking successful!
```

```catala-test-inline
$ catala Interpret -s A
[WARNING] The structure "Bar" is never used; maybe it's unnecessary?

┌─⯈ tests/test_struct/good/same_name_fields.catala_en:7.23-7.26:
└─┐
7 │ declaration structure Bar:
  │                       ‾‾‾
  └─ Article
[RESULT] Computation successful! Results:
[RESULT] x = Foo { -- f: 1 }
[RESULT] y = 1
```
```catala-test-inline
$ catala Interpret_Lcalc -s A --avoid_exceptions --optimize
[WARNING] The structure "Bar" is never used; maybe it's unnecessary?

┌─⯈ tests/test_struct/good/same_name_fields.catala_en:7.23-7.26:
└─┐
7 │ declaration structure Bar:
  │                       ‾‾‾
  └─ Article
[RESULT] Computation successful! Results:
[RESULT] x = Foo { -- f: 1 }
[RESULT] y = 1
```
