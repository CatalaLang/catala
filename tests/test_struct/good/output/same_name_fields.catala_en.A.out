[RESULT] Computation successful! Results:
[RESULT] x = Foo {"f": 1}
[RESULT] y = 1
